src/consensus/block_producer.rs

```diff
@@ -165,11 +170,21 @@ where
                 None
             };
 
-            // cap timeout for each slice to `DELTA_BLOCK`
-            // makes sure optimistic block production yields before timeout would expire
-            let time_for_slice = duration_left.min(self.delta_block);
+            let time_for_slice = if slice_index == SliceIndex::first() {
+                // make sure first slice is produced on time
+                // TODO: this can be made more accurate, only needed if production of first slice
```

Comment:
For my understanding: it can still be the case that the leader has not seen parent ready but the other nodes have seen it.  In that case, the way the code is currently would make sense, i.e. the leader should not delay producing the first slice longer.